app/Repositories/Project/ProjectRepository.php

```diff
@@ -3,19 +3,97 @@
 namespace App\Repositories\Project;
 
 use App\Models\Project;
+use Storage;
 use App\Repositories\BaseRepository;
 use App\Repositories\Project\ProjectRepositoryInterface;
 use Illuminate\Support\Collection;
+use App\Repositories\Activity\ActivityRepositoryInterface;
+use Illuminate\Http\Request;
 
-class ProjectRepository extends BaseRepository implements ProjectRepositoryInterface
+class ProjectRepository extends BaseRepository implements ProjectRepositoryInterface 
 {
+    
+    private $activityRepository;
     /**
      * ProjectRepository constructor.
      *
      * @param Project $model
      */
-    public function __construct(Project $model)
+    public function __construct(Project $model,ActivityRepositoryInterface $activityRepository) 
     {
+        $this->activityRepository = $activityRepository;
         parent::__construct($model);
     }
+
+    public function clone(Request $request,Project $project) 
```

Comment:
Formatting issue.
Secondly, this 'clone' function must also be defined in the ProjectRepositoryInterface.